javascript/unit-tests/KeySet.test.ts

```diff
@@ -0,0 +1,191 @@
+
+import { sleep } from "./test_utils";
+import { GinkInstance, Bundler, IndexedDbStore, Value } from "../implementation";
+import { ensure, matches } from "../implementation/utils"
+import { KeySet } from "../implementation";
+
+test('add and has basic data', async function() {
+    // set up the objects
+    const store = new IndexedDbStore('test1', true);
+    const instance = new GinkInstance(store);
+    const ks = await instance.createKeySet();
+
+    // add a value
+    await ks.add("key1");
+
+    // check that the result exists in the database
+    ensure(await ks.has("key1"));
+
+    const myKey = new Uint8Array(3);
+    myKey[0] = 94;
+    myKey[2] = 255;
+
+    await ks.add(myKey);
+    ensure(await ks.has(myKey));
+});
+
+test('delete, pop, and size work as intended', async function() {
+    const store = new IndexedDbStore('test2', true);
+    const instance = new GinkInstance(store);
+    const ks = await instance.createKeySet();
+
+    await ks.add("key1");
+    ensure(await ks.has("key1"));
+    ensure(await ks.size() === 1);
+
+    await ks.delete("key1");
+    ensure(await ks.size() === 0);
```

Comment:
Also need to test that ks.has("key1") is false after the delete.

And as mentioned elsewhere pop should not be part of this interface.